File: 2016/5/5.js
"use strict";

const fs = require("fs");

let kryptomelding = fs.readFileSync("kryptomelding.txt", "utf-8");

console.time();

// Alle mulige romertall i denne oppgaven
const romans = ["0", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X", "XI", "XII", "XIII", "XIV", "XV", "XVI", "XVII", "XVIII", "XIX", "XX", "XXI", "XXII", "XXIII", "XXIV", "XXV", "XXVI"];

// Bytt ut romer med desimal
for (let i in romans) {
	kryptomelding = kryptomelding.replace(new RegExp("\\b" + romans[i] + "\\b", "g"), i);
}

// Parse som JSON
kryptomelding = JSON.parse(kryptomelding);

let result = [];

// Legg sammen første og siste tall
while (kryptomelding.length > 0) {
	result.push(kryptomelding.shift() + kryptomelding.pop());
}

// Konverter til bokstaver
let a = "a".charCodeAt(0)-1;
result = result.map(n => String.fromCharCode(a+n)).join("");

// Print resultatet
console.log(result);
console.timeEnd();

File: 2016/5/5.hs
import System.IO (readFile)
import Data.List (elemIndex)
import Data.Maybe (fromJust)
import Data.Char (chr, ord)

romans :: [String]
romans = ["0", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X", "XI", "XII", "XIII", "XIV", "XV", "XVI", "XVII", "XVIII", "XIX", "XX", "XXI", "XXII", "XXIII", "XXIV", "XXV", "XXVI"];

main = do
    contents <- readFile "kryptomelding.txt"
    let romanTokens = words $ filter ((/=) ',') $ tail $ init contents
        numTokens = map (fromJust . flip elemIndex romans) romanTokens
        cleartext = take (length numTokens `div` 2) $
                    map (chr . (+) (ord 'a' - 1)) $
                    zipWith (+) numTokens (reverse numTokens)
    putStrLn cleartext
